### ISSUE TITLE
Print layout - Legend items with expression return -1

### ISSUE THREAD

**Issue by EnFiore** [qgis/QGIS#40154]
Hi,
I have print layout with legend items that have these expressions:

1) `@symbol_label || ' tot: ' ||  @symbol_count ` 
and this:
2) `@symbol_label || ' length: ' || round(sum($length,filter:=  "cat" = @symbol_label ),2)`

and in layout the expressions work right showing correct values.

But when I export as image, the first expression return -1.
I attach some image to show the issue and a project to replicate it.

![lyt_value](https://user-images.githubusercontent.com/11796614/99509556-31a22b00-2986-11eb-87ce-5fcc2f58975f.PNG)
![img_export](https://user-images.githubusercontent.com/11796614/99509559-32d35800-2986-11eb-9244-ea291de8653a.jpg)

[issue_legend_items_qgis.zip](https://github.com/qgis/QGIS/files/5558932/issue_legend_items_qgis.zip)

**Comment by gioman** [comment]
@EnFiore QGIS version? Operating system?

**Comment by gioman** [comment]
Those expressions are not editable in your layout (the button is greyed out), is that expected?

**Comment by roya0045** [comment]
@gioman you must select the layers for the button to be activated.
@EnFiore This is a known issue with the rendering pipeline of expressions in the legend, see #34476 which should address things in part, though I still get some strange glitch visually. I'm not sure if this is my old MXE build acting up or a resource issue. I am awaiting a first review from @nyalldawson to improve the PR and make the rendering/export more reliable.

**Comment by gioman** [comment]
> @gioman you must select the layers for the button to be activated.

@roya0045 I know, here it doesn't work anyway.

**Comment by roya0045** [comment]
Just for information, currently the expressions are evaluated after the image is exported, this is why such things are broken for the time being in master and recent releases.

**Comment by roya0045** [comment]
@gioman works fine for me, might be an issue with your build?

**Comment by gioman** [comment]
> @gioman works fine for me, might be an issue with your build?

@roya0045 maybe, no custom build here. QGIS 3.16 straight out of qgis.org repos for ubuntu 20.04

**Comment by roya0045** [comment]
@gioman care to post a screenshot? I'm curious, my 3.16 build is fine with expressions,

**Comment by EnFiore** [comment]
> 
> 
> @EnFiore QGIS version? Operating system?


QGIS version 3.10.11-A Coruña 
Operating system: Windows 10

**Comment by EnFiore** [comment]
> 
> 
> Those expressions are not editable in your layout (the button is greyed out), is that expected?

@gioman no, in my project, the button is active and expression editables.

**Comment by scm181** [comment]
Hello,
I have the same problem (using WIN 10 and QGIS 3.16.2), in my case it's showing n/a instead of the symbol_count.
Is there a hint or solution, I can use?
Best regards

**Comment by roya0045** [comment]
@scm181 can you shared the project and related data?

**Comment by Pedro-Murteira** [comment]
Still a valid issue on QGIS 3.22.3. (Windows 10)

**Comment by kelm-solutions** [comment]
Hi, any further developments or solutions to that issue? Ver 3.28-4 still has this "-1" @symbol_count value while exporting to jpg/pdf
